### PR TITLE
ci: tsan with wallet

### PR DESCRIPTION
**Pull request by maflcko**

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK: more TSan coverage is better :)

**Comment by practicalswift** [comment]
ACK fa7e002d520d8390f3ff4b0383cfdfc14713355d -- patch looks correct and Travis is happy

**Review by hebasto — APPROVED** [review]
ACK fa7e002d520d8390f3ff4b0383cfdfc14713355d, I have reviewed the code and it looks OK, I agree it can be merged.

File: test/sanitizer_suppressions/tsan [hunk 1]
```diff
@@ -12,17 +12,22 @@ mutex:CConnman::ThreadOpenAddedConnections
...
 race:ThreadImport
+race:WalletBatch::WriteHDChain
 race:zmq::*
 race:bitcoin-qt
 # deadlock (TODO fix)
 deadlock:CConnman::ForNode
+deadlock:CConnman::GetNodeStats
+deadlock:CChainState::ConnectTip
 deadlock:UpdateTip
+deadlock:wallet_tests::CreateWalletFromFile
```
Comment (ryanofsky) [inline test/sanitizer_suppressions/tsan:30]: Thanks for this. There's discussion about this case in https://github.com/bitcoin/bitcoin/issues/19049

**Pull request merged**